crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -430,33 +594,125 @@ impl<'a> DocstringSections<'a> {
     }
 }
 
+/// Parse the entries in a "Parameters" section of a docstring.
+///
+/// Attempts to parse using the specified [`SectionStyle`], falling back to the other style if no
+/// entries are found.
+fn parse_parameters(content: &str, style: Option<SectionStyle>) -> Vec<&str> {
+    match style {
+        Some(SectionStyle::Google) => parse_parameters_google(content),
+        Some(SectionStyle::Numpy) => parse_parameters_numpy(content),
+        None => {
+            let entries = parse_parameters_google(content);
+            if entries.is_empty() {
+                parse_parameters_numpy(content)
+            } else {
+                entries
+            }
+        }
+    }
+}
+
+/// Parses Google-style "Args" sections of the form:
+///
+/// ```python
+/// Args:
+///     a (int): The first number to add.
+///     b (int): The second number to add.
+/// ```
+fn parse_parameters_google(content: &str) -> Vec<&str> {
+    let mut entries: Vec<&str> = Vec::new();
+    // Find first entry to determine indentation
+    let mut indentation = None;
+    for potential in content.lines() {
+        if potential.find(':').is_none() {
+            continue;
+        };
+        indentation = Some(&potential[..potential.len() - potential.trim_start().len()]);
+        break;
+    }
+    let Some(indentation) = indentation else {
+        return entries;
+    };
+    for potential in content.lines() {
+        if let Some(entry) = potential.strip_prefix(indentation) {
+            if entry
+                .chars()
+                .next()
+                .is_some_and(|first_char| !first_char.is_whitespace())
+            {
+                let Some((before_colon, _)) = entry.split_once(':') else {
+                    continue;
+                };
+                if let Some(param) = before_colon.split_whitespace().next() {
+                    entries.push(param.trim_start_matches('*'));
+                }
+            }
+        }
+    }
+    entries
+}
+
+/// Parses NumPy-style "Parameters" sections of the form:
+///
+/// ```python
+/// Parameters
+/// ----------
+/// a : int
+///     The first number to add.
+/// b : int
+///     The second number to add.
+/// ```
+fn parse_parameters_numpy(content: &str) -> Vec<&str> {
+    let mut entries: Vec<&str> = Vec::new();
+    let mut lines = content.lines();
+    let Some(dashes) = lines.next() else {
+        return entries;
+    };
+    let indentation = &dashes[..dashes.len() - dashes.trim_start().len()];
```

Comment:
Why do we not try to find the indentation similar to Google style docstring? Or, why do we need to _find_ the indentation in Google style docstring? Is it that Google style can contain blank lines while NumPy style cannot?

Reply:
We do it this way because the dashes in the numpy style docs directly give us the indentation, and the args should use the same indent. There is no dashes in google style so we have to infer it from the args.

I don't really understand your follow up question.

Reply:
What I mean is why don't we use the first line to determine the indentation in Google style docstring similar to what we're doing here for NumPy style docstring. This is just a minor nit so feel free to ignore.

Reply:
Oh ya makes sense. I implemented the suggestion.